### PR TITLE
AB#89873 Add State access for sites

### PR DESCRIPTION
**Pull request by dawid-pawulski**
Added structure for access state in Sites.
Local authorities will be set to true (it is always accessible).
Other state will be filled by @Antek012 and @rafalgajewski

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs [hunk 1]
```diff
@@ -26,7 +26,33 @@ public async Task<SiteWorkflowState> NextState(Trigger trigger)
...
+            SiteWorkflowState.LocalAuthoritySearch => true,
+            SiteWorkflowState.LocalAuthorityResult => true,
+            SiteWorkflowState.LocalAuthorityConfirm => true,
+            SiteWorkflowState.LocalAuthorityReset => true,
+            SiteWorkflowState.PlanningStatus => true,
+            SiteWorkflowState.NationalDesignGuide => true,
+            SiteWorkflowState.PlanningDetails => true,
+            SiteWorkflowState.BuildingForHealthyLife => true,
+            SiteWorkflowState.LandRegistry => true,
+            _ => throw new ArgumentOutOfRangeException(nameof(state), state, null),
```
Comment (Carq) [inline src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:54]: Maybe by default should be always `true`?

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs (hunk 1 above)
Comment (jjankowski87) [reply src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:54]: I think it should be explicitly specified that state can be accessed. We can add unit tests which goes through all enum values and checks that exception is not thrown

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs (hunk 1 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:54]: So the default should be `false`. Because there is no need to throw an exception.

Comment (Antek012) [reply src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:54]: @dawid-pawulski can we skip merging this changes?
On my local branch I am working on changes for this file.

**Pull request closed without merging**